x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/http/StreamingHttpResultPublisherTests.java

```diff
@@ -84,6 +86,52 @@ public void testFirstResponseCallsListener() throws IOException {
         assertThat("Listener's onResponse should be called when we receive a response", latch.getCount(), equalTo(0L));
     }
 
+    /**
+     * When we receive an http response with an entity with no content
+     * Then we call the listener
+     * And we queue the initial payload
+     */
+    public void testEmptyFirstResponseCallsListener() throws IOException {
+        var latch = new CountDownLatch(1);
+        var listener = ActionListener.<Flow.Publisher<HttpResult>>wrap(
```

Comment:
FYI there is a utility function `ActionTestUtils.assertNoFailureListener()` that creates a failing listener for you

https://github.com/elastic/elasticsearch/blob/b15b76b8694e6b9f837b4f3b65bc16ff9b4b497f/test/framework/src/main/java/org/elasticsearch/action/support/ActionTestUtils.java#L72